### PR TITLE
Add devops recipe: github actions

### PR DESCRIPTION
**Pull request by zzstoatzz**
add separate `docker build` and `prefect deployment build` workflows and associated dockerfile in a new `dockerfiles` folder

## Description
- `minimal-prefect-deployment-build.yaml`: run `prefect deployment build` if our flow file changes 
- `docker-build-push-gcp-artifact-registry.yaml`: build and push a docker image for our flow to Google Artifact Registry if our Dockerfile or dependencies change

## Type of change
- [x] This change requires a documentation update

## New Recipe Checklist

- [x] My PR is in the format of `Add <project-name> recipe`
- [x] My recipe is reproducible and explains everything needed to run successfully. If my code has external dependencies, I make mention of them.
- [x] My code is easily understandable and/or well-commented.
- [x] If my recipe requires a new category (e.g. creating a monitoring/ folder in devops/), I have encapsulated my project within its own subfolder so that others can add their own recipes as well.

### CODE REVIEW

File: devops/github-actions/docker-build-push-gcp-artifact-registry.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+on:
```
Comment (serinamarie) [inline devops/github-actions/docker-build-push-gcp-artifact-registry.yaml:1]: Might be worth giving this a `name` ✨

File: devops/github-actions/docker-build-push-gcp-artifact-registry.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  push:
```
Comment (serinamarie) [inline devops/github-actions/docker-build-push-gcp-artifact-registry.yaml:2]: Might be worth adding a`workflow_dispatch` option for different work styles? Then with comments like

```
on:
    workflow_dispatch:  # build new image manually
    push:  # or automatically build upon change to dependencies
        branches:
            - "main"
        paths: 
            - "Dockerfile" 
            - "requirements.txt"
```

File: devops/github-actions/docker-build-push-gcp-artifact-registry.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          registry: 'us-docker.pkg.dev'
+          username: 'oauth2accesstoken'
+          password: '${{ steps.auth.outputs.access_token }}'
+
+      - name: Push to container registry
+        run: |
+          docker push ${{ env.FLOW_IMAGE }}
+      - name: Output Success
+        run: echo "Successfully built and pushed ${{ env.FLOW_IMAGE }}!"
```
Comment (serinamarie) [inline devops/github-actions/docker-build-push-gcp-artifact-registry.yaml:46]: [Adding a job summary](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary) would look really polished. 🧐 💅 

`run: echo "Successfully built and pushed ${{ env.FLOW_IMAGE }}! :rocket:" >> $GITHUB_STEP_SUMMARY`

**Comment by serinamarie** [comment]
Looks great! I only commented on one of the workflows but I think the same comments could apply to both.

**Review by serinamarie — APPROVED** [review]
Thanks for adding a name, summary output and workflow_dispatch option to the other Actions as well. I really appreciate it-- LGTM!

Edit: Because there is a new folder being made called `dockerfiles/` in `devops/`, the change to the directory tree should be evaluated by @prefectcboyd or @gabcoyne.

**Review by prefectcboyd — APPROVED** [review]
Lgtm

**Pull request merged**